foundationdb/init-foundationdb.sh

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env bash
+# init-foundationdb.sh script ensures that FoundationDB server
+# is installed and configured on the machine.
+
+set -eu
+
+SCRIPT_DIR="$( realpath --strip "$( dirname "${BASH_SOURCE[0]}" )" )"
+
+source "$(dirname ${SCRIPT_DIR})/common.sh"
+
+############################
+### Arguments processing ###
+############################
+
+usage() {
+    echo "Usage: ${0} \\"
+    echo "         --cluster-file-content <STR> --host-to-listen-on <HOST> \\"
+    echo "         [--data-dir <PATH>] [--fdbcli-command <COMMAND>]"
+    echo
+    echo "Options:"
+    echo "   --cluster-file-content <STR>  fdb.cluster file content."
+    echo "   --host-to-listen-on <HOST>    Host name or IP to listen on."
+    echo "   --data-dir <PATH>             FDB data dir path (default: /var/lib/foundationdb/data/)."
+    echo "   --fdbcli-command <COMMAND>    fdbcli command to execute after cluster initialization."
+    exit 1
+}
+
+CLUSTER_FILE_CONTENT=
+HOST_TO_LISTEN_ON=
+DATA_DIR=/var/lib/foundationdb/data/
+FDBCLI_COMMAND=
+
+# Let's get the command line arguments.
+while [ $# -gt 0 ]; do
+    if [ "${1}" = "--cluster-file-content" ]; then
+        shift
+        if [ -z "${1}" ]; then
+            usage
+        fi
+        CLUSTER_FILE_CONTENT="${1}"
+        shift
+        continue
+    fi
+    if [ "${1}" = "--host-to-listen-on" ]; then
+        shift
+        if [ -z "${1}" ]; then
+            usage
+        fi
+        HOST_TO_LISTEN_ON="${1}"
+        shift
+        continue
+    fi
+    if [ "${1}" = "--data-dir" ]; then
+        shift
+        if [ -z "${1}" ]; then
+            usage
+        fi
+        DATA_DIR="${1}"
+        shift
+        continue
+    fi
+    if [ "${1}" = "--fdbcli-command" ]; then
+        shift
+        if [ -z "${1}" ]; then
+            usage
+        fi
+        FDBCLI_COMMAND="${1}"
+        shift
+        continue
+    fi
+    usage
+done
+
+if [ -z "${CLUSTER_FILE_CONTENT}" ] || [ -z "${HOST_TO_LISTEN_ON}" ]; then
+    usage
+fi
+
+CLUSTER_SERVERS=$(echo "${CLUSTER_FILE_CONTENT}" | awk -F '@' '{ print $2 }' | tr '\r\n' ' ' | sed 's/,/ /g' )
+HOST_SERVER_IDS=
+for server_address in ${CLUSTER_SERVERS} ; do
+  host=$(echo "${server_address}" | awk -F ':' '{ print $1 }')
+  port=$(echo "${server_address}" | awk -F ':' '{ print $2 }')
+  if [ "${host}" = "${HOST_TO_LISTEN_ON}" ] ; then
+    HOST_SERVER_IDS+="${port} "
+  fi
+done
+
+if [ -z "${HOST_SERVER_IDS}" ] ; then
```

Comment:
Cluster files should only contain a list of coordination servers, so the absence of the current machine's IP from a cluster file should not indicate that the current machine should not be part of the cluster.

The typical procedure for adding additional machines to an existing cluster is taking a cluster file from a machine in the current cluster and starting fdb on another machine with that cluster file. I think it's a bit unusual for a cluster file to be defined before the cluster exists since the idea is that they are meant to be descriptive of a given cluster rather than prescriptive about how a cluster should be.

A multinode setup seems like it would be a job for an external system coordinating the work.